lib/budget_issue_hook.rb

```diff
@@ -22,12 +23,27 @@ def view_issues_show_details_bottom(context = { })
   # * :project => Current project
   #
   def view_issues_form_details_bottom(context = { })
-    if context[:project].module_enabled?('budget_module')
-      select = context[:form].select :deliverable_id, Deliverable.find_all_by_project_id(context[:project], :order => 'subject ASC').collect { |d| [d.subject, d.id] }, :include_blank => true 
-      return "<p>#{select}</p>"
-    else
-      return ''
+    unless context[:project].module_enabled?("budget_module")
+      return ""
+    end
+
+    deliverables = Deliverable.where(project_id: context[:project]).order("subject ASC")
+    hidden_ids = deliverables.where(hidden: true).pluck(:id)
```

Comment:
Using `pluck` here causes an extra db query, because we already load the relation in `deliverables.map`.
This means this should be rewritten to something like:
```ruby
deliverables.select(&:hidden?).map(&:id)
```